sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+
+// @public
+export type ListSparkStatementsResponse = SparkStatementCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: SparkStatementCollection;
+    };
+};
+
+// @public
+export const logger: import("@azure/logger").AzureLogger;
+
+// @public (undocumented)
+export interface OperationResponse {
+    _response: HttpResponse;
+}
+
+export { PipelineOptions }
+
+// @public
+export type PluginCurrentState = "Preparation" | "ResourceAcquisition" | "Queued" | "Submission" | "Monitoring" | "Cleanup" | "Ended" | string;
+
+// @public (undocumented)
+export type ResetSparkSessionTimeoutOptions = OperationOptions;
+
+// @public
+export type SchedulerCurrentState = "Queued" | "Scheduled" | "Ended" | string;
+
+// @public (undocumented)
+export interface SparkBatchJob {
+    appId?: string;
+    appInfo?: {
+        [propertyName: string]: string;
+    };
+    artifactId?: string;
+    errors?: SparkServiceError[];
+    id: number;
+    jobType?: SparkJobType;
+    // (undocumented)
+    livyInfo?: SparkBatchJobState;
+    logLines?: string[];
+    name?: string;
+    plugin?: SparkServicePlugin;
+    result?: SparkBatchJobResultType;
+    scheduler?: SparkScheduler;
+    sparkPoolName?: string;
+    state?: string;
+    submitterId?: string;
+    submitterName?: string;
+    tags?: {
+        [propertyName: string]: string;
+    };
+    workspaceName?: string;
+}
+
+// @public
+export interface SparkBatchJobCollection {
+    from: number;
+    sessions?: SparkBatchJob[];
+    total: number;
+}
+
+// @public (undocumented)
+export interface SparkBatchJobOptions {
+    // (undocumented)
+    archives?: string[];
+    // (undocumented)
+    arguments?: string[];
+    // (undocumented)
+    artifactId?: string;
+    // (undocumented)
+    className?: string;
+    configuration?: {
+        [propertyName: string]: string;
+    };
+    // (undocumented)
+    driverCores?: number;
+    // (undocumented)
+    driverMemory?: string;
+    // (undocumented)
+    executorCores?: number;
+    // (undocumented)
+    executorCount?: number;
+    // (undocumented)
+    executorMemory?: string;
+    // (undocumented)
+    file: string;
+    // (undocumented)
+    files?: string[];
+    // (undocumented)
+    jars?: string[];
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    pythonFiles?: string[];
+    tags?: {
+        [propertyName: string]: string;
+    };
+}
+
+// @public
+export type SparkBatchJobResultType = "Uncertain" | "Succeeded" | "Failed" | "Cancelled" | string;
+
+// @public (undocumented)
+export interface SparkBatchJobState {
+    currentState?: string;
+    deadAt?: Date;
+    // (undocumented)
+    jobCreationRequest?: SparkRequest;
+    notStartedAt?: Date;
+    recoveringAt?: Date;
+    runningAt?: Date;
+    startingAt?: Date;
+    successAt?: Date;
+    terminatedAt?: Date;
+}
+
+// @public (undocumented)
+export class SparkClient {
```

Comment:
Do you mean Create operation need to support LRO? I think we can stay current function, because that api has been used for a not short time at different languages, we don't get feedback need to support LRO. 
Another question, can we keep this createSparkSession for now and add a new beginCreateSparkSession supoortted LRO in the future?

---
In reply to: [521603316](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521603316) [](ancestors = 521603316)

Reply:
One of the principles of the SDK is consistency, so I would hesitate to offer multiple ways to initiate / monitor / cancel a long running operation other than to use our LRO primitive. 

I would think it would be simpler to have a consistent abstraction, especially since you should be able to change the internal workings of the LRO as needed.